packages/language-server/src/linting.ts

```diff
@@ -0,0 +1,63 @@
+import {
+  findEndPosition,
+  parserWrapper,
+  validateSyntax,
+} from '@neo4j-cypher/language-support';
+import debounce from 'lodash.debounce';
+import { join } from 'path';
+import { Diagnostic, TextDocumentChangeEvent } from 'vscode-languageserver';
+import { TextDocument } from 'vscode-languageserver-textdocument';
+import workerpool from 'workerpool';
+import { LinterTask, LintWorker } from './lint-worker';
+
+const pool = workerpool.pool(join(__dirname, 'lint-worker.js'), {
+  minWorkers: 2,
```

Comment:
Why do we need 2 workers? I think you explained me this already.

I tried including a `console.log` of the `pool.stats()` at the beginning of the linting here and at the beginning of the semantic linting and I can see 1 occupied worker, 1 idle one at most.


```
{totalWorkers: 2, busyWorkers: 1, idleWorkers: 1, pendingTasks: 0, activeTasks: 1}
```

Reply:
The intention is to always have an idle worker available, so that we can avoid the startup times of spawning a new thread. My reasoning is that when one thread is busy, we can use the idle one without startup cost and then the old busy thread will be terminated and respawned so it's ready for the next call

I've not measured the start up time/performance gains though